### PR TITLE
Check if netstat or iproute2 is available

### PR DESCRIPTION
**Pull request by cblecker**
**What this PR does / why we need it**:
In https://github.com/kubernetes/kubernetes/pull/59655, I incorrectly assumed that the kubetest image would have net-tools. It doesn't. This checks and uses `ss` from iproute2, *or* `netstat` from net-tools. The kubetest image does appear to have iproute2.

**Release note**:
```release-note
NONE
```

### CODE REVIEW

**Comment by dims** [comment]
@cblecker did you mean to push the pipe error message commit to this PR?

File: hack/lib/etcd.sh [hunk 1]
```diff
@@ -28,7 +28,16 @@ kube::etcd::validate() {
   }
 
   # validate etcd port is free
-  if netstat -nat | grep "[\.:]${ETCD_PORT:?} .*LISTEN" >/dev/null 2>&1; then
+  local port_check_command
+  if which ss &> /dev/null && ss -Version | grep 'iproute2' &> /dev/null; then
```
Comment (stevekuznetsov) [inline hack/lib/etcd.sh]: I'd use `command -v` as it's apparently kosher for implementations of `which` to [not set exit status](https://stackoverflow.com/a/677212) and look at loads of things other than installed binaries. On my machine, `which` is:

```
$ type which
which is aliased to `(alias; declare -f) | /usr/bin/which --tty-only --read-alias --read-functions --show-tilde --show-dot'
```

**Comment by cblecker** [comment]
/hold cancel

PTAL @dims @stevekuznetsov :)

**Comment by stevekuznetsov** [comment]
/hold

for @dims

**Comment by dims** [comment]
/hold cancel

LGTM 👍 

```
ubuntu@testvm-1:~/go/src/k8s.io/kubernetes$ export ETCD_PORT=2379
ubuntu@testvm-1:~/go/src/k8s.io/kubernetes$ ss -nat | grep "LISTEN" | grep "[\.:]${ETCD_PORT:?}"
LISTEN     0      128    127.0.0.1:2379                     *:*
ubuntu@testvm-1:~/go/src/k8s.io/kubernetes$ netstat -nat | grep "LISTEN" | grep "[\.:]${ETCD_PORT:?}"
tcp        0      0 127.0.0.1:2379          0.0.0.0:*               LISTEN
```

**Review by dims — APPROVED** [review]

**Pull request merged**